contract/test/utilities/exceptions.ts

```diff
@@ -0,0 +1,25 @@
+import { assert } from 'chai'
+
+function isException (error) {
+  const strError = error.toString()
+  return strError.includes('invalid opcode') || strError.includes('invalid JUMP') || strError.includes('revert')
+}
+
+function ensureException (error) {
+  assert(isException(error), error.toString())
+}
+
+async function expectFailure (call) {
```

Comment:
Also checkout `chai-as-promised` - I have used it before for some other stuff. Not sure if there is a more up-to-date version

Reply:
removed as wasn't using this (was copied from old ens codebase)